airflow/auth/managers/fab/api/auth/backend/kerberos_auth.py

```diff
@@ -0,0 +1,153 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+# Copyright (c) 2013, Michael Komitee
```

Comment:
Why the extra copyright here? Which of the code below falls under it?

Reply:
Agree. I was tempted to remove it but I played it safe and kept it as is

Reply:
Happy to remove it unless there is a good reason

Reply:
But do you know what it is for or what code it applies to?

Reply:
It looks like that license has been there since 2016, added in d5ac6bd9d0257683d8b2d0b200a513b2f43ce5dc by @bolkedebruin

Reply:
I'll remove it unless a good reason is provided. Looks like very old. Thanks for looking into it

Reply:
With PMC 🎩  on (and bear with me please).

IMHO - we should keep the licence in kerberos_auth.py. 

Moving the file around does not remove the authorship/change the licences for those source files we evidently copied from another project (flask-kerberos) (unless we remove that code from the original contributed code).

This is stated in the text of the BSD licence itself - it obliges us to do so:

> 1. Redistributions of source code must retain the above copyright notice, this
   list of conditions and the following disclaimer.

We are redistributing the source code by making it available to our users to download - both via https://downloads.apache.org/airflow/ but also via `pypi` packages (which contain the sources). So we should keep the licence in the source code.

And we are also following the second point from the licence:

> # 2. Redistributions in binary form must reproduce the above copyright notice,
# this list of conditions and the following disclaimer in the documentation
# and/or other materials provided with the distribution.

Because we keep https://github.com/apache/airflow/blob/main/licenses/LICENSE-flask-kerberos.txt as part of our packages.

Licences/Compliance with the ASF policy https://www.apache.org/legal/resolved.html  is one of the most important things for the PMC members to keep an eye on.

And yes, I very much sympathise with "which part of the code it applies to" @o-nikolas and it might not be clear. Technically we should **track** it (via git history) and keep the copyright wherever the code is moved. In this case it's rather simple - just put it in "kerberos_auth.py". This was the original intention of the change as I see it. It would have been better if we just "vendored-it-in" in "_vendor/" directory - but we did not, originally.

There are lengthy discussions about it in the ASF Legal JIRA (for example https://issues.apache.org/jira/browse/LEGAL-649?page=com.atlassian.jira.plugin.system.issuetabpanels%3Aall-tabpanel ) which you likely do not want to read. But just to be on the safe side (none of us is a lawyer) we should just keep it there.

I know, boring and not obvious. but just sticking to keeping the licence is best approach in this case.

Reply:
Sounds good